src/pages/resultsView/fusion/ResultViewFusionMapperStore.ts

```diff
@@ -0,0 +1,212 @@
+ * version 3, or (at your option) any later version.
+ *
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+import {
+    CancerStudy,
+    Gene,
+    MolecularProfile,
+    SampleIdentifier,
+    StructuralVariant
+} from '../../../shared/api/generated/CBioPortalAPI';
+import { cached, labelMobxPromises, MobxPromise } from "mobxpromise";
+import { computed } from 'mobx';
+import { remoteData } from '../../../shared/api/remoteData';
+import { fetchEnsemblTranscriptsByEnsemblFilter, fetchPfamDomainData } from '../../../shared/lib/StoreUtils';
+import { EnsemblTranscript, PfamDomain } from '../../../shared/api/generated/GenomeNexusAPI';
+import * as _ from 'lodash';
+import { EnsemblTranscriptExt, StructuralVariantExt } from '../../../shared/model/Fusion';
+import ResultViewFusionMapperDataStore from './ResultViewFusionMapperDataStore';
+
+export const FUS_GENE_COLORSCALE = ['#097AFF', '#3B9CFF', '#28a9ff', '#84B7DF', '#92C8E8', '#68ACEA', '#4eb7d9'];
+export const REF_GENE_COLORSCALE = ['#88C542', '#99c556', '#a0cf31', '#7ba237', '#47cb4a', '#008828', '#005623'];
+
+export class ResultViewFusionMapperStore {
+
+    /**
+     * @param {Gene} gene
+     * @param {MobxPromise<{[p: string]: CancerStudy}>} studyIdToStudy
+     * @param {MobxPromise<{[p: string]: MolecularProfile}>} molecularProfileIdToMolecularProfile
+     * @param {MobxPromise<SampleIdentifier[]>} samples
+     * @param {StructuralVariant[]} fusions
+     */
+    constructor(public gene: Gene,
+                public studyIdToStudy: MobxPromise<{ [studyId: string]: CancerStudy }>,
+                public molecularProfileIdToMolecularProfile: MobxPromise<{ [molecularProfileId: string]: MolecularProfile }>,
+                public samples: MobxPromise<SampleIdentifier[]>,
+                public fusions: StructuralVariant[]) {
+        labelMobxPromises(this);
+    }
+
+
+    /**
+     * Returns fusion data with label added as new attribute composed of site1 and site2 genes.
+     * @type {MobxPromiseUnionTypeWithDefault<StructuralVariantExt[]>}
+     */
+    readonly fusionData = remoteData({
+        invoke: async () => {
+            return (<StructuralVariantExt[]> this.fusions || []).map(f => {
+                // generate fusion label made of site1 and site2 genes
+                f.label = `${f.site1HugoSymbol}-${f.site2HugoSymbol}`;
```

Comment:
I think its better to use an object for this because gene names can have hyphens in them. its the most edgey of edge case but I don't think its a big deal and its better safe than sorry. what do you think?